src/AppController.m

```diff
@@ -333,14 +336,18 @@ -(void)installSleepHandler
  * Install a handler to notify of changes in the scripts folder.
  * The handler is a code block which triggers a refresh of the scripts menu
  */
--(void)installScriptsFolderWatcher
-{
-	NSURL * path = [NSURL fileURLWithPath:[Preferences standardPreferences].scriptsFolder];
-	_events = [[CDEvents alloc] initWithURLs:@[path]
-                                       block:^(CDEvents *watcher, CDEvent *event) {
-										   // triggers a refresh of the scripts.menu
-                                           [self initScriptsMenu];
-                                       }];
+- (void)installScriptsFolderWatcher {
+    NSURL *path = [NSURL fileURLWithPath:Preferences.standardPreferences.scriptsFolder];
+    self.directoryMonitor = [[DirectoryMonitor alloc] initWithDirectories:@[path]];
+
+    NSError *error = nil;
+    [self.directoryMonitor startAndReturnError:&error eventHandler:^{
+        typeof(self) __weak weakSelf = self;
```

Comment:
Do you know why some people suggest you create the weakSelf outside of the block? e.g. http://stackoverflow.com/a/23628986/2691531

Did Instruments pick up a leak having the weakSelf created inside the block?